### PR TITLE
test: Adding tests for LDContextBuilder

### PR DESCRIPTION
**Pull request by ActuallyConnor**
**Requirements**

- [x] I have added test coverage for new or changed functionality
- [x] I have followed the repository's [pull request submission guidelines](../blob/main/CONTRIBUTING.md#submitting-pull-requests)
- [x] I have validated my changes against all supported platform versions

**Related issues**

N/A

**Describe the solution you've provided**

Improve test coverage and test directly [LDContextBuilder.php](https://github.com/launchdarkly/php-server-sdk/blob/main/src/LaunchDarkly/LDContextBuilder.php)

**Describe alternatives you've considered**
This class was tested indirectly previously but contained logic that would benefit from testing directly.

**Additional context**

N/A

### CODE REVIEW

File: tests/LDContextBuilderTest.php [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+                null,
+                false,
+                ['attribute-name' => 'value'],
+                null,
+                null,
+                null
+            ),
+            $context
+        );
+    }
```
Comment (keelerm84) [inline tests/LDContextBuilderTest.php:165]: I think all of these are redundant with tests we already have.

File: tests/LDContextBuilderTest.php [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+
+    /**
+     * @dataProvider trySetDataProvider
```
Comment (keelerm84) [inline tests/LDContextBuilderTest.php:168]: I will comment on the provider which tests I think are providing new value.

File: tests/LDContextBuilderTest.php [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        $this->assertEquals($expectedIsSet, $isSet);
+
+        $context = $this->builder->build();
+        $this->assertEquals(
+            $expectedLdContext,
+            $context
+        );
+    }
+
+    public function testPrivate(): void
```
Comment (keelerm84) [inline tests/LDContextBuilderTest.php:183]: This test is also redundant.

File: tests/LDContextBuilderTest.php [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+                    AttributeReference::fromLiteral('attributeName')
+                ],
+                null,
+                null
+            ),
+            $context
+        );
+    }
+
+    public function testPrivateEmptyAttributeRefs(): void
```
Comment (keelerm84) [inline tests/LDContextBuilderTest.php:212]: We don't have coverage for this exact scenario. I would be okay including this test if you can move it with the others and update to match the testing style in that file.

File: tests/LDContextBuilderTest.php [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+                    LDContext::DEFAULT_KIND,
+                    'some-context-key',
+                    null,
+                    false,
+                    ['attribute-name' => 'value'],
+                    null,
+                    null,
+                    null
+                )
+            ],
```
Comment (keelerm84) [inline tests/LDContextBuilderTest.php:340]: These are redundant.

File: tests/LDContextBuilderTest.php [hunk 6]
```diff
@@ -0,0 +1,358 @@
...
+            'attributes name value' => [
```
Comment (keelerm84) [inline tests/LDContextBuilderTest.php:341]: We don't have a test checking setting a custom attribute to null. You could probably just fold that into the `testBuilderSetCustomAttributes` test.

**Review by keelerm84 — CHANGES_REQUESTED** [review]
Thank you for your interest and in taking the time to contribute!

> This class was tested indirectly previously but contained logic that would benefit from testing directly.

I'm not sure which tests you are referring to. If you look at `LDContextTest.php`, most of the tests interact with the context builder (via `LDContext::builder`). As a result, I think several of the tests in this PR are redundant with what we currently have.

However, there are a couple of tests which I think we did miss. If you want to move those tests into that file, I would be happy to include those contributions. I've tried to note which ones to keep and which to save. 

LMK if you have any questions!

**Comment by ActuallyConnor** [comment]
@keelerm84 I've applied the requested changes to this PR

**Review by keelerm84 — APPROVED** [review]

**Comment by keelerm84** [comment]
Thank you for addressing this feedback, and once again, thank you for taking the time to contribute to this project!

**Pull request merged**